cmd/analyze/main.go

```diff
@@ -100,10 +100,22 @@ func dynamicAnalysis(pkg *pkgecosystem.Pkg) {
 
 	if *uploadFileWriteInfo != "" {
 		bucket, path := parseBucketPath(*uploadFileWriteInfo)
-		err := resultstore.New(bucket, resultstore.BasePath(path)).Save(ctx, pkg, results.FileWrites)
+		err := resultstore.New(bucket, resultstore.BasePath(path)).Save(ctx, pkg, results.FileWritesSummary)
 		if err != nil {
 			log.Fatal("Failed to upload file write analysis results to blobstore", "error", err)
 		}
+		for _, writeBufferPathsArray := range results.FileWriteBufferPaths {
+			for _, writeBufferPath := range writeBufferPathsArray {
```

Comment:
Can you add some logging similar to https://github.com/ossf/package-analysis/pull/609 for profiling how long these take to upload?

Reply:
To give more context: GCS can be a little slow if we upload lots of small files. We may want to have some goroutine-based parallel uploading here if it does turn out to be too slow.

Reply:
I ran this locally and clocked it at 26.5 seconds. This feels a bit long to me if we need to wait for that every run of dynamic analysis, but I'm not sure what a reasonable threshold is.

Reply:
Took a look through some of the logs from running worker. This one looks pretty bad.
Dynamic analysis uploaded file write buffers	{"ecosystem": "pypi", "name": "rapidpe-rift-pipe", "version": "0.0.7.dev20230223", "dynamic_analysis_upload_file_write_buffers_duration": "11m30.483768089s"}

Reply:
That certainly is not good. Can you try zipping them up and uploading them as a single file and seeing how much faster it is?

Reply:
I can't seem to reproduce as bad of performance as the 11 minutes from above anymore but here's some stats from running that package and version on local analysis:

Small individual file writes: 
{"ecosystem": "pypi", "name": "rapidpe-rift-pipe", "version": "0.0.7.dev20230223", "error": null, "dynamic_analysis_phase_duration": "2m25.341110532s"}

Writing one zip file with the zip writer.
Writes duration finished	{"ecosystem": "pypi", "name": "rapidpe-rift-pipe", "version": "0.0.7.dev20230223", "error": null, "dynamic_analysis_phase_duration": "44.012550917s"}

I took a look at ~100 packages in the worker analysis logs and longest duration I saw was the following: 
{"ecosystem": "pypi", "name": "pretrained-backbones-unet", "version": "0.0.1", "write_data_upload_duration": "1m18.155039659s"}

It still does take some time to run, but definitely an improvement. I left the logs in the code. I think it could be helpful but right now there's so many logs I feel like it could just end up getting lost? I currently track time to write both the file writes summary .json file as well as the zip file with the write buffers but the bulk of the time is spent on the zip file.